packages/ptokens-assets-evm/src/lib/index.ts

```diff
@@ -1,183 +1,159 @@
-import { sha256 } from '@noble/hashes/sha256'
-import { AbiEvent, AbiEventParameter } from 'abitype'
-import BigNumber from 'bignumber.js'
-import { FactoryAddress, NetworkId } from 'ptokens-constants'
-import { AssetInfo } from 'ptokens-entities'
+import { AbiEvent } from 'abitype'
+import { Operation } from 'ptokens-entities'
+import { stringUtils } from 'ptokens-helpers'
 import {
   Log,
-  toBytes,
-  toHex,
-  encodeAbiParameters,
   decodeEventLog,
-  getEventSelector,
+  toEventSelector,
   Chain,
   TransactionReceipt,
   createPublicClient,
   http,
+  decodeAbiParameters,
+  parseAbiParameters,
+  slice,
+  hexToNumber,
+  hexToString,
+  size,
+  hexToBigInt,
+  concat,
+  pad,
+  sha256,
 } from 'viem'
-import { arbitrum, bsc, gnosis, polygon } from 'viem/chains'
 
-import factoryAbi from '../abi/PFactoryAbi'
-import pNetworkHubAbi from '../abi/PNetworkHubAbi'
+import pNetworkAdapterAbi from '../abi/PNetworkAdapterAbi'
 import { pTokensEvmProvider } from '../ptokens-evm-provider'
 
-const events = pNetworkHubAbi.filter(({ type }) => type === 'event') as AbiEvent[]
 export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
 
-const networkIdToViemChain: Record<NetworkId, Chain> = {
-  [NetworkId.ArbitrumMainnet]: arbitrum,
-  [NetworkId.BscMainnet]: bsc,
-  [NetworkId.GnosisMainnet]: gnosis,
-  [NetworkId.PolygonMainnet]: polygon,
-}
-
-export const formatAddress = (_address: string): `0x${string}` =>
-  _address.startsWith('0x') ? (_address as `0x${string}`) : (('0x' + _address) as `0x${string}`)
-
-export const getViemChain = (networkId: NetworkId): Chain => {
-  switch (networkId) {
-    case NetworkId.PolygonMainnet:
-      return polygon
-      break
-    default:
-      throw new Error(`${networkId} is not supported by ptokens.js`)
-  }
-}
-
-export function onChainFormat(_amount: BigNumber.Value, _decimals: number): BigNumber {
-  return BigNumber(_amount).multipliedBy(BigNumber(10).pow(_decimals))
-}
-
-export function offChainFormat(_amount: BigNumber.Value, _decimals: number) {
-  return BigNumber(_amount).dividedBy(BigNumber(10).pow(_decimals))
-}
-
 export enum EVENT_NAMES {
-  OPERATION_QUEUED = 'OperationQueued',
-  OPERATION_EXECUTED = 'OperationExecuted',
-  OPERATION_CANCELLED = 'OperationCancelled',
-  USER_OPERATION = 'UserOperation',
+  SWAP = 'Swap',
+  SWAP_NATIVE = 'SwapNative',
```

Comment:
It doesn't exists
https://github.com/pnetwork-association/pnetwork-v4/blob/master/solidity/src/interfaces/IAdapter.sol#L25
```suggestion
```